### DISCUSSION TITLE
Cell type selection and visualization

### DISCUSSION THREAD

**Discussion by mzabolocki** [petersenpeter/CellExplorer#67] [General]
Hello, 

firstly - thank you for creating such a wonderful tool! 

I have spike-sorted a .dat file using kilosort and outputted the rez file. I have now run the cell explorer tutorial, creating a session metadata struct from a basepath directed to a folder containing both the rez.mat file and raw file. 

When running the cell metrics pipeline 'ProcessCellMetrics' using the session struct as input I get a display of only  4 isolated waveforms (pyramidal neurons, narrow and wide interneurons, unknown). When I visualize the cell metrics in CellExplorer, the identified cell types have very similar/identical waveforms and in addition the custom group, cell type separation, and tSNE dimensionality reduction plots only display these 4 data points. 

There are 1,000,000 + spikes detected during the Kilosort spike-sorting step. Looking at your online youtube tutorial, and also matlab tutorial via the github repo, I can see some stark differences in the output. 

Notably, the recordings are from human brain cortical organoids - would this be an issue with the cell type classification, or is it simply the way the data has been processed and inputted into the CellExplorer GUI? Or am I missing some key parameter inputs (e.g. a metadata file...)? 

Please find some example outputs below from a single 64 channel recording. Thankyou very much in advance! :) 

<img width="677" alt="cellexplorer_issue3" src="https://user-images.githubusercontent.com/89680425/187201395-891a8e97-4abf-4e42-ba72-8ea76efb219a.PNG">
<img width="893" alt="cellexplorer_issue2" src="https://user-images.githubusercontent.com/89680425/187201026-cffc41cb-7c30-4b7e-9941-5b0d70d8f791.PNG">
<img width="904" alt="cellexplorer_issue1" src="https://user-images.githubusercontent.com/89680425/187201038-519f07c7-c5b1-417d-a698-60c5483a5185.PNG">

**Comment by petersenpeter** [comment]
Hi 
Were you expecting more cells and did you curate your cells? (One of your units looks to be a real unit, the others don't.) CellExplorer expects manually curated cells if you set "phy" as spike sorting format. What I mean here is that it will only import cells labeled as good in phy. If you import directly from KiloSort, perhaps it uses labels assigned by Kilosort to select these four units, but I would have to see your data to know better. Could you share a session with me (petersen.peter@gmail.com)? This type of error is typically due to wrong session parameters. I would need the raw dat file, and your session metadata (basename.session.mat and original metadata), and your kilosort rez file, if this is where you import your units from.

/Peter

**Reply by mzabolocki** [reply]
Thank you very much Peter for the help! I'll send you an email today :)
